### PR TITLE
Add CLI tests for `contract` and `wallet` commands

### PR DESCRIPTION
**Pull request by fyrchik**
Related #1576.
Tests for `neo-go node` are not done here.

### CODE REVIEW

File: cli/executor_test.go [hunk 1]
```diff
@@ -132,7 +132,7 @@ func (e *executor) GetTransaction(t *testing.T, h util.Uint256) (*transaction.Tr
 func (e *executor) getNextLine(t *testing.T) string {
 	line, err := e.Out.ReadString('\n')
 	require.NoError(t, err)
-	return line
+	return strings.TrimSuffix(line, "\n")
```
Comment (roman-khimov) [inline cli/executor_test.go:135]: Commit says `add tests`, this is something different from that.

File: cli/smartcontract/smart_contract.go [hunk 2]
```diff
@@ -56,7 +57,7 @@ var (
 		Name:  "out",
 		Usage: "file to put JSON transaction to",
 	}
-	forceFlag = cli.StringFlag{
+	forceFlag = cli.BoolFlag{
```
Comment (roman-khimov) [inline cli/smartcontract/smart_contract.go:60]: And this one too.

File: pkg/compiler/codegen.go [hunk 3]
```diff
@@ -879,8 +879,12 @@ func (c *codegen) Visit(node ast.Node) ast.Visitor {
 				for _, p := range n.Args[1:] {
 					params = append(params, c.scTypeFromExpr(p))
 				}
-				name := constant.StringVal(tv.Value)
-				c.emittedEvents[name] = append(c.emittedEvents[name], params)
+				// Sometimes event name is stored in a var.
+				// Skip in this case.
```
Comment (roman-khimov) [inline pkg/compiler/codegen.go:883]: Separate commit.

File: cli/wallet/legacy.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+			Parameters []wallet.ContractParam `json:"parameters"`
+			Deployed   bool                   `json:"deployed"`
+		} `json:"contract"`
+		Locked  bool `json:"lock"`
+		Default bool `json:"isdefault"`
+	}
+)
+
+// newWalletV2FromFile reads NEO2 wallet from file.
+// This is used as read-only, no are supported on returned wallet.
```
Comment (roman-khimov) [inline cli/wallet/legacy.go:35]: no what are supported?

File: cli/smartcontract/smart_contract.go (hunk 2 above)
Comment (fyrchik) [reply cli/smartcontract/smart_contract.go:60]: Fixed.

File: pkg/compiler/codegen.go (hunk 3 above)
Comment (fyrchik) [reply pkg/compiler/codegen.go:883]: Fixed.

File: cli/executor_test.go (hunk 1 above)
Comment (fyrchik) [reply cli/executor_test.go:135]: Fixed.

File: cli/contract_test.go [hunk 5]
```diff
@@ -16,25 +16,85 @@ import (
...
+	defer os.RemoveAll(tmpDir)
+
+	e := newExecutor(t, false)
+	defer e.Close(t)
+
+	t.Run("no path is provided", func(t *testing.T) {
+		e.RunWithError(t, "neo-go", "contract", "init")
+	})
+	t.Run("invalid path", func(t *testing.T) {
+		e.RunWithError(t, "neo-go", "contract", "init", "--name", string([]byte{0}))
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply cli/contract_test.go:31]: Fixed.

File: cli/wallet/legacy.go (hunk 4 above)
Comment (fyrchik) [reply cli/wallet/legacy.go]: `operations`, fixed.

**Comment by fyrchik** [comment]
Actually, fixed #1589 right now.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**